### PR TITLE
[Tests-only] prepare ocis API tests

### PR DESCRIPTION
**Pull request by individual-it**
## Description
1. prepare the test infrastructure to run API tests on OCIS
2. skip all tests that don't work on OCIS and adjust others to make them work on oC10 and OCIS

tests that are skipped on OCIS and don't have an issue number are skipped because of:
- no internal user management in OCIS
- cannot create a "browser session"

that are not really issues, so not issue was created in github

## Related Issue

- part of https://github.com/owncloud/ocis-reva/issues/23

## Motivation and Context
test OCIS

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/TestHelpers/WebDavHelper.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public static function getFileIdForPath(
 		$response = self::makeDavRequest(
 			$baseUrl, $user, $password, "PROPFIND", $path, null, $body
 		);
-		\preg_match('/\<oc:fileid\>(\d+)\<\/oc:fileid\>/', $response, $matches);
+		\preg_match('/\<oc:fileid\>(.*)\<\/oc:fileid\>/', $response, $matches);
```
Comment (individual-it) [inline tests/TestHelpers/WebDavHelper.php]: the fileid on OCIS, has also other characters

File: tests/TestHelpers/OcisHelper.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * ownCloud
+ *
+ * @author    Artur Neumann <artur@jankaritech.com>
+ * @copyright Copyright (c) 2017 Artur Neumann artur@jankaritech.com
```
Comment (phil-davis) [inline tests/TestHelpers/OcisHelper.php]:
```suggestion
 * @copyright Copyright (c) 2020 Artur Neumann artur@jankaritech.com
```

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 3]
```diff
@@ -8,6 +8,7 @@ Feature: download file
     Given user "user0" has been created with default attributes and without skeleton files
     And user "user0" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt"
     And user "user0" has uploaded file with content "Welcome this is just an example file for developers." to "/welcome.txt"
+    Given user "user1" has been created with default attributes and without skeleton files
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]:
```suggestion
    And user "user1" has been created with default attributes and without skeleton files
```

File: tests/acceptance/features/apiFavorites/favorites.feature [hunk 4]
```diff
@@ -92,12 +103,13 @@ Feature: favorite
     When user "user1" moves file "/shared/shared_file.txt" to "/taken_out.txt" using the WebDAV API
     Then user "user1" in folder "/" should have favorited the following elements
       | /taken_out.txt |
-    Examples:
+    Examples:m
```
Comment (phil-davis) [inline tests/acceptance/features/apiFavorites/favorites.feature]: random "m"
```suggestion
    Examples:
```

File: tests/acceptance/features/apiShareCreateSpecial/createShareWithDisabledUser.feature [hunk 5]
```diff
@@ -1,4 +1,4 @@
-@api @TestAlsoOnExternalUserBackend @files_sharing-app-required
+@api @TestAlsoOnExternalUserBackend @files_sharing-app-required @skipOnOcis
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecial/createShareWithDisabledUser.feature]: does this need to have an issue?

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 6]
```diff
@@ -237,6 +343,7 @@ Feature: sharing
       | 1               | 100             |
       | 2               | 200             |
 
+  @skipOnOcis
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: does this need to have an issue number?

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 7]
```diff
@@ -266,6 +374,7 @@ Feature: sharing
...
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: does this need to have an issue number?

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 8]
```diff
@@ -298,6 +408,7 @@ Feature: sharing
       | 2               | 200             |
 
   @issue-35484
+  @skipOnOcis
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: does this need to have an ocis issue number?

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 9]
```diff
@@ -8,6 +8,7 @@ Feature: download file
     Given user "user0" has been created with default attributes and without skeleton files
     And user "user0" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt"
     And user "user0" has uploaded file with content "Welcome this is just an example file for developers." to "/welcome.txt"
+    And user "user1" has been created with default attributes and without skeleton files
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature]: why do we need user1?

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 10]
```diff
@@ -1941,9 +1959,20 @@ public function cleanupGroup($group) {
...
+			$requestingPassword = $this->getPasswordForUser($requestingUser);
+		} else {
+			$requestingUser = $this->getAdminUsername();
+			$requestingPassword = $this->getAdminPassword();
+		}
 		$fullUrl = $this->getBaseUrl() . "/ocs/v2.php/cloud/users/$user";
 		$this->response = HttpRequestHelper::get(
-			$fullUrl, $this->getAdminUsername(), $this->getAdminPassword()
+			$fullUrl, $requestingUser,
+			$requestingPassword
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php]: Put all parameters on one line, or 3 lines

File: tests/acceptance/features/apiShareCreateSpecial/createShareWithDisabledUser.feature (hunk 5 above)
Comment (individual-it) [reply tests/acceptance/features/apiShareCreateSpecial/createShareWithDisabledUser.feature]: there is no internal user management in OCIS, so users cannot be disabled

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
please don't merge till the other side works in https://github.com/owncloud/ocis-reva/pull/59

File: tests/acceptance/features/apiShareCreateSpecial/createShareWithDisabledUser.feature (hunk 5 above)
Comment (individual-it) [reply tests/acceptance/features/apiShareCreateSpecial/createShareWithDisabledUser.feature]: there is no internal user management in OCIS, so users cannot be disabled

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 11]
```diff
@@ -1129,6 +1130,7 @@ public function userGetsSizeOfFileUsingTheWebdavApi($user, $resource) {
 	 */
 	public function theSizeOfTheFileShouldBe($size) {
 		$responseXml = HttpRequestHelper::getResponseXml($this->response);
+		echo $responseXml->asXML();
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]: remove debug line

File: tests/acceptance/features/apiWebdavOperations/refuseAccess.feature [hunk 12]
```diff
@@ -21,6 +21,21 @@ Feature: refuse access
       | old         |
       | new         |
 
+  @smokeTest @skipOnOcV10
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavOperations/refuseAccess.feature]: is there an issue?

**Pull request merged**